### PR TITLE
[BACKLOG-24377]

### PR DESCRIPTION
**Pull request by graimundo**
- General dependency cleanup
- Made pentaho-dataservice feature more independent from the standard pentaho deployment in order to use them in a pdi-core/foundry environment
- DataServiceResolver(s) are now injected via OSGi services

@pentaho/millenniumfalcon @joaormf please review

### CODE REVIEW

File: assemblies/dataservice/pom.xml [hunk 1]
```diff
@@ -14,6 +14,43 @@
   <name>Pentaho Community Edition Project: ${project.artifactId}</name>
   <description>Pentaho Data Services Server and Client bundled</description>
 
+
+
+  <dependencyManagement>
+    <dependencies>
+      <!-- rxJava is provided via karaf custom.properties because of AEL ??? -->
```
[the remark this answers is not in the document]
Comment (graimundo) [reply assemblies/dataservice/pom.xml]: All karaf assemblies at the moment have rxJava provided via custom.properties. It was AEL that put it there, but all assemblies are like that regardless of having AEL.

File: impl/src/main/resources/OSGI-INF/blueprint/services.xml [hunk 2]
```diff
@@ -71,20 +71,24 @@
...
+      <argument ref="serviceCacheFactory"/>
+      <argument value="${transient.service.loglevel}"/>
+    </bean>
+
+    <bean id="metaStoreResolverBean" class="org.pentaho.di.trans.dataservice.resolvers.MetaStoreResolver">
+      <argument ref="repositoryLocator"/>
+      <argument ref="context"/>
+    </bean>
+
+    <service ref="transientResolverBean" interface="org.pentaho.di.trans.dataservice.resolvers.DataServiceResolver" ranking="1"/>
```
[the remark this answers is not in the document]
Comment (graimundo) [reply impl/src/main/resources/OSGI-INF/blueprint/services.xml]: I believe it is like that in order to maintain the previous behaviour. @nantunes Please correct me if I'm wrong.

File: assemblies/dataservice/src/main/feature/feature.xml [hunk 3]
```diff
@@ -2,6 +2,21 @@
...
+
+  </feature>
+
+  <!-- This feature is here as a hack because the pentaho-metaverse feature from the
+  standard pentaho-server repository loads half the world.
+   We still need to confirm if metaverse needs to bring anything else along. -->
+  <feature name="pentaho-metaverse-core" version="${project.version}">
+    <feature>pentaho-tinkerpop-gremlin</feature>
+
+    <configfile finalname="/etc/pentaho.metaverse.cfg">
```
[the remark this answers is not in the document]
Comment (graimundo) [reply assemblies/dataservice/src/main/feature/feature.xml]: I don't disagree having a variable for the karaf etc configuration folder. AFAIK we are currently using the /etc directly. In the marketplace plugin we're doing it like that and it is working on windows:
https://github.com/pentaho/marketplace/blob/master/pentaho-marketplace-assembly/src/main/feature/feature.xml#L4
https://github.com/pentaho/marketplace/blob/master/pdi-marketplace-assembly/src/main/feature/feature.xml#L5

File: impl/src/main/resources/OSGI-INF/blueprint/services.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (nantunes) [reply impl/src/main/resources/OSGI-INF/blueprint/services.xml]: Yes. It previously was at the top of the list, before MetaStoreResolver.

File: assemblies/dataservice/src/main/feature/feature.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (joaormf) [reply assemblies/dataservice/src/main/feature/feature.xml]: @graimundo it seems to be done everywhere... so it should work :)

File: impl/src/main/resources/OSGI-INF/blueprint/services.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (joaormf) [reply impl/src/main/resources/OSGI-INF/blueprint/services.xml]: I see in the docs (http://aries.apache.org/modules/blueprint.html) that ranking is used to select a reference when multiple are available (default is 0, and the higher ranking is preferred). But it doesn't state anything about the order where they appear in a a reference-list.

We could be having a "lucky" side-effect?

**Pull request merged**